support-frontend/assets/pages/paper-subscription-landing/paperSubscriptionLandingPage.tsx

```diff
@@ -52,13 +52,29 @@ const pageQaId = 'qa-paper-subscriptions';
 function PaperLandingPage({
 	productPrices,
 	promotionCopy,
-}: PaperLandingContentPropTypes) {
+	participations,
+}: PaperLandingPropTypes) {
 	const sanitisedPromoCopy = getPromotionCopy(promotionCopy);
-	const fulfilment: PaperFulfilmentOptions = window.location.pathname.includes(
-		'delivery',
-	)
-		? HomeDelivery
-		: Collection;
+
+	const isNationalDeliveryAbTestVariant =
+		participations.nationalDelivery === 'variant';
+
+	const [shouldRedirectToHomeDelivery, setShouldRedirectToHomeDelivery] =
+		useState<boolean>(isNationalDeliveryAbTestVariant);
+
+	if (
+		shouldRedirectToHomeDelivery &&
+		!window.location.pathname.includes('delivery')
+	) {
+		window.history.replaceState({}, '', paperSubsUrl(true));
+		setShouldRedirectToHomeDelivery(false);
+	}
```

Comment:
So the logic here is: if we’ve landed on the paper landing page and the national delivery AB test participation is in the variant, change the user’s URL to include `delivery` at the end, and register that we’ve done that so we don’t do it twice – right?

I dislike the weirdness of landing on `…/paper/delivery` (after redirection) and clicking the other tab to go to `…/paper`, but I’m not sure there’s a straightforward alternative. To my mind the fulfilment option makes more sense as a query parameter (e.g. `…/paper?fulfilmentOption=HomeDelivery`), and then if it’s not set we’d just assume a value, the AB test changing the order of the tabs and which value we assume. But I don’t know this code well enough to be confident that there isn’t a good reason it’s in the path, and I’m not sure I want us to get sucked in to changing it now.

What do you think?

Reply:
yep: that is the logic 👍 

I agree with the weirdness, although that was the existing functionality. Could definitely spend more time refactoring the page to use query params instead, but I don't the code either

Reply:
Let’s leave it